client/web/admin/src/views/System/Cdns/Index.vue

```diff
@@ -0,0 +1,257 @@
+<template>
+  <b-container
+    class="pt-2 pb-3"
+  >
+    <c-content-header
+      :title="$t('title')"
+    />
+    <b-card
+      body-class="p-0"
+      header-class="border-bottom"
+      footer-class="border-top d-flex flex-wrap flex-fill-child gap-1"
+      class="shadow-sm"
+    >
+      <div class="align-items-center gap-1 p-3">
+        <b-button
+          variant="primary"
+          size="lg"
+          @click="newCDN()"
+        >
+          {{ $t('cdns-provider.add') }}
+        </b-button>
+      </div>
+
+      <b-table
+        :items="providerItems"
+        :fields="cdnProviderFields"
+        head-variant="light"
+        hover
+        class="mb-0"
+      >
+        <template #cell(provider)="{ item }">
+          {{ item.provider || item.tag }}
+        </template>
+
+        <template #cell(editor)="{ item }">
+          <c-input-confirm
+            v-if="item.delete"
+            :icon="item.deleted ? ['fas', 'trash-restore'] : undefined"
+            @confirmed="item.delete()"
+          />
+
+          <b-button
+            variant="link"
+            @click="openEditor(item.editor)"
+          >
+            <font-awesome-icon
+              :icon="['fas', 'wrench']"
+            />
+          </b-button>
+        </template>
+      </b-table>
+
+      <b-modal
+        id="modal-cdn"
+        v-model="modal.open"
+        :title="modal.title"
+        scrollable
+        size="lg"
+        title-class="text-capitalize"
+        @ok="modal.updater(modal.data)"
+      >
+        <component
+          :is="modal.component"
+          v-model="modal.data"
+        />
+        <template #modal-footer="{ ok, cancel}">
+          <b-button
+            size="sm"
+            variant="danger"
+            @click="deleteCDN(modal.index)"
+          >
+            Delete
```

Comment:
Translation strings. Also the delete button should only be visible for existing code snippets.
It should also be aligned left.

Make sure its an input confirm aswell